src/nvim/eval.c

```diff
@@ -15392,7 +15416,7 @@ static void f_setpos(typval_T *argvars, typval_T *rettv)
  */
 static void f_setqflist(typval_T *argvars, typval_T *rettv)
 {
-  set_qf_ll_list(NULL, &argvars[0], &argvars[1], rettv);
+  set_qf_ll_list(NULL, &argvars[0], &argvars[1], &argvars[2], rettv);
```

Comment:
Same problem here as in `f_setloclist()`. We are missing argument checks to prevent  calling `set_qf_ll_list()` with arguments that do not exist.